test/errors.test.ts

```diff
@@ -42,6 +42,20 @@ describe('errors', () => {
         expect(ensureErrorType(actualError)).to.equal(actualError);
     });
 
+    it('readonly error', () => {
+        class ReadOnlyError extends Error {
+            readonly message = 'a readonly message';
```

Comment:
The TypeScript `readonly` keyword is pretty useless tbh - it disappears when compiled to JavaScript. Here is the typical way to make something actually readonly:
```typescript
class ReadOnlyError extends Error {
    get message() {
        return 'a readonly message';
    }
}
```

You shouldn't need to do this: `Object.defineProperty(actualError, 'message', { writable: false });`

Reply:
It's a little surprising how unhelpful `readonly` actually is. I have updated as suggested.